mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -100,6 +100,7 @@ pub struct S3ClientConfig {
     read_backpressure: bool,
     initial_read_window: usize,
     network_interface_names: Vec<String>,
+    telemetry_callback: Option<Arc<dyn OnTelemetry>>,
```

Comment:
I've used dynamic dispatch here since I don't think you want this type to leak into all the other types (i.e. S3CrtClient, S3CrtClientInner, S3ClientConfig, etc)